pkgs/applications/editors/spacevim/default.nix

```diff
@@ -0,0 +1,58 @@
+{ makeWrapper, vim_configurable, vimPlugins, fetchFromGitHub, writeTextDir
+, stdenv, spacevim_toml ? "" }:
+with stdenv;
+let
+  # At the moment, there is no way to write TOML given a attrset
+  # so the solution here is to allow a custom TOML file to be provided.
+  # The pkg includes a sensible TOML file for those that don't specify one.
+  config = if builtins.stringLength spacevim_toml == 0 then
+    "${builtins.readFile ./init.toml}"
+  else
+    spacevim_toml;
```

Comment:
How about merging `spacevim_toml` and `init.toml` together so that a user doesn't have to rewrite the entire config when having to change a few details.

Also, I'm wondering if it might be better to accept an attr-set and generate TOML-config from that. Bonus points for writing some docs (in `doc/builders/packages`).

Reply:
Well i'm not sure about a merge since that sounds like making deletes challening.
I would love to use something like `nix2toml` but I couldn't find a good solution.
(maybe remarshal ?)

Reply:
Is there any reason against converting `init.toml` into a Nix attr-set? Then you can merge it together with another attr-set using `lib.recursiveUpdate`.

In the end you'd only have to render a single attr-set to `toml`. Sth. like this seem to be implemented in `nixos/modules/virtualisation/containers.nix` already.

Reply:
Done in latest rebase